src/realm/parser/driver.cpp

```diff
@@ -809,7 +809,7 @@ std::unique_ptr<Subexpr> ConstantNode::visit(ParserDriver* drv, DataType hint)
                 ret = new Value<Decimal128>(Decimal128(text));
             }
             else {
-                ret = new Value<int64_t>(strtol(text.c_str(), nullptr, 0));
+                ret = new Value<int64_t>(strtoll(text.c_str(), nullptr, 0));
```

Comment:
If you wanted to not worry about locale, you could use https://github.com/realm/realm-core/blob/7f3067113b51110bf2ea37de8caeaded0b7b7b7d/src/realm/util/from_chars.hpp#L24 here.

Reply:
Cool, I didn't know we had this! It looks like it doesn't support a prefix of "+" which we do allow here though.